### PR TITLE
🧹 Clean up auth code

### PR DESCRIPTION
**Pull request by jackieo5023**
This PR stack will ultimately clean up and centralize all of the auth code so that it's more easily understandable.  This will be done using providers and contexts so that we don't have to pass the auth props down multiple component levels.

This PR specifically sets up the logic for the `useAuth` hook that can be used to get auth variables anywhere in the app.

I tested various actions throughout the app (login, logout, view admin stuff) to make sure I didn't break anything.

### CODE REVIEW

File: client/src/utils/use-auth.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// @flow
+
+// code inspired from https://usehooks.com/useAuth/
+import * as React from 'react';
```
Comment (n3a9) [inline client/src/utils/use-auth.js:4]: I think `import *` isn't super good, maybe if there's a way to import exactly what we want

File: client/src/utils/use-auth.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          },
+        ),
+      ]).then(vals => setAuthRoles(Object.keys(vals[0].roles).concat('')));
+    }
+    fetchData();
+  }, []);
+
+  const authRoleIsEquivalentTo = useCallback(
+    role => {
+      if (authRole == null) {
```
Comment (n3a9) [inline client/src/utils/use-auth.js:53]: `if (!authRole)` or `if (authRole === null)`

File: client/src/utils/use-auth.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        return null;
+      }
+      if (authRoles == null) {
```
Comment (n3a9) [inline client/src/utils/use-auth.js:56]: maybe combine the two if statements just to condense more. super nit though

**Review by n3a9 — APPROVED** [review]
lgtm

File: client/src/utils/use-auth.js (hunk 2 above)
Comment (jackieo5023) [reply client/src/utils/use-auth.js:53]: flow doesn't like :(

File: client/src/utils/use-auth.js (hunk 1 above)
Comment (jackieo5023) [reply client/src/utils/use-auth.js:4]: flow also doesn't like

**Pull request merged**